File: solutions/225-implement-stack-using-queues/implement-stack-using-queues.php

class MyStack {
    private $q = [];

    public function push($x) {
        array_push($this->q, $x);
        for ($i = 0; $i < count($this->q) - 1; $i++) {
            array_push($this->q, array_shift($this->q));
        }
    }

    public function pop() {
        return array_shift($this->q);
    }

    public function top() {
        return $this->q[0];
    }

    public function empty() {
        return empty($this->q);
    }
}

File: solutions/225-implement-stack-using-queues/implement-stack-using-queues.rs
use std::collections::VecDeque;

pub struct MyStack {
    q: VecDeque<i32>,
}

impl MyStack {
    pub fn new() -> Self {
        MyStack { q: VecDeque::new() }
    }

    pub fn push(&mut self, x: i32) {
        self.q.push_back(x);
        for _ in 0..self.q.len() - 1 {
            let val = self.q.pop_front().unwrap();
            self.q.push_back(val);
        }
    }

    pub fn pop(&mut self) -> i32 {
        self.q.pop_front().unwrap()
    }

    pub fn top(&self) -> i32 {
        *self.q.front().unwrap()
    }

    pub fn empty(&self) -> bool {
        self.q.is_empty()
    }
}